src/main/java/games/strategy/engine/framework/headlessGameServer/AvailableGames.java

```diff
@@ -99,18 +99,23 @@ private static void populateAvailableGames(final Map<String, URI> availableGames
     return Arrays.asList(files);
   }
 
-  private static void populateFromDirectory(final File mapDir, final Map<String, URI> availableGames,
-      final Set<String> availableMapFolderOrZipNames, final Set<String> mapNamePropertyList) {
+  private static void populateFromDirectory(
+      final File mapDir,
+      final Map<String, URI> availableGames,
+      final Set<String> availableMapFolderOrZipNames,
+      final Set<String> mapNamePropertyList) {
     final File games = new File(mapDir, "games");
     if (!games.exists()) {
       // no games in this map dir
       return;
     }
-    for (final File game : games.listFiles()) {
-      if (game.toURI() != null && game.isFile() && game.getName().toLowerCase().endsWith("xml")) {
-        final boolean added = addToAvailableGames(game.toURI(), availableGames, mapNamePropertyList);
-        if (added) {
-          availableMapFolderOrZipNames.add(mapDir.getName());
+    if (games.listFiles() != null) {
```

Comment:
It looks like if you don't extract the result of `games.listFiles()` to a local and use that in the if check and for-each loop, you have a TOCTOU.  Not likely to happen, but...

Note also that, per the [JDK](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#listFiles--), `File#listFiles()` can return `null` if an I/O error occurs, not just if the descriptor does not represent a directory.